pkg/errors/errors.go

```diff
@@ -0,0 +1,110 @@
+// Package errors introduces utilities to wrap error with additional metadata
+// so that the error has proper context when it is logged.
+// The idea was borrowed from: https://medium.com/@oberonus/context-matters-advanced-error-handling-techniques-in-go-b470f763c7ec
+package errors
+
+import (
+	"errors"
+)
+
+// errWithMetadata represents an error with attached metadata
+type errWithMetadata struct {
+	// err is the original error
+	err error
+	// metadata is the container for error context
+	metadata []string
+}
+
+// Error returns the original error message,
+// ensuring compatibility with the standard error interface.
+func (w *errWithMetadata) Error() string {
+	return w.err.Error()
+}
+
+// Unwrap returns the original error that was wrapped with errWithMetadata instance
+// It makes the errWithMetadata compatible with the standard error unwrapping mechanism
+func (w *errWithMetadata) Unwrap() error {
+	return w.err
+}
+
+type Metadata []string
```

Comment:
why not `[]any`?

Reply:
If we use `[]any`, this will allow following usage:
```go
return errhelper.WithMetadata(err, errMetadata.Extend("key1", "value1"))
```
This will call `WithMetadata()` function with only one key of type `Metadata{"key1", "value1"}` which is not what we want.

Using `[]string` enforce that you have expanded the returned slice and use following instead:
```go
return errhelper.WithMetadata(err, errMetadata.Extend("key1", "value1")...)
```